cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -631,6 +631,18 @@ public void testAppRedeployKeepsData() throws Exception {
     Assert.assertEquals("value1", myTableManager3.get().get(new Get("key1", "column1")).getString("column1"));
   }
 
+  @Test(timeout = 60000L)
+  public void testFlowletMetricsReset() throws Exception {
+    ApplicationManager appManager = deployApplication(DataSetInitApp.class);
+    FlowManager flowManager = appManager.startFlow("DataSetFlow");
+    RuntimeMetrics flowletMetrics = RuntimeStats.getFlowletMetrics("DataSetInitApp", "DataSetFlow", "Consumer");
+    flowletMetrics.waitForProcessed(1, 5, TimeUnit.SECONDS);
+    flowManager.stop();
+    RuntimeStats.resetAll();
```

Comment:
use RuntimeMetrics.getProcessed before and after - before it should be > 0, after it should be 0. Using waitfor is hacky